src/javascript/utils.js

```diff
@@ -69,15 +69,63 @@ export function loadScript(url) {
 
 
 /**
- * Escapes a potentially unsafe HTML string.
- * @param {string} str An string that may contain HTML entities.
+ * Escapes a potentially unsafe HTML string. Makes no attempt to preserve
+ * entities like &nbsp;.
+ * @param {string} str A string that may contain HTML entities.
  * @return {string} The HTML-escaped string.
  */
-export function escapeHtml(str) {
-  const div = document.createElement('div');
-  div.appendChild(document.createTextNode(str));
-  return div.innerHTML;
-}
+export const escapeHtml = str => str
```

Comment:
Hmmm, I think the recommended way of doing this these days is here:
https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

And if you're going to manually escape it (and you're already using lodash) couldn't you just use that?

Reply:
Does... that help us? DOMParser allows us to convert a string of HTML into a HTMLDocument instance, and doesn't seem to offer any aid in parsing. The code I replaced basically works, but I replaced it because the old version was very heavy (creating and destroying a whole node), when the only thing you need to do for correctness is replace the 5 characters. There's no risk of script injection, because the <script> tags would also be replaced. I can explicitly document that it's not safe for substituting content INTO scripts, which would be a crazy thing to do, anyway.

I was not aware that lodash shipped with an html escape function! Will use that instead.

Reply:
SGTM.

I'm mistaken, DOMParser is a good way to remove HTML from a string. What you have is fine. (BTW, the old was probably fine too. I found a jsperf showing browsers could do that like 800,000 times a second, so the perf hit is minimal).

The only gotcha I know of with the `_.escape` method is in cases where an attribute isn't quoted.